ethereum/core/src/main/java/org/hyperledger/besu/ethereum/vm/operations/SLoadOperation.java

```diff
@@ -15,32 +15,40 @@
 package org.hyperledger.besu.ethereum.vm.operations;
 
 import org.hyperledger.besu.ethereum.core.Account;
-import org.hyperledger.besu.ethereum.core.Gas;
-import org.hyperledger.besu.ethereum.vm.AbstractOperation;
+import org.hyperledger.besu.ethereum.vm.EVM;
 import org.hyperledger.besu.ethereum.vm.GasCalculator;
 import org.hyperledger.besu.ethereum.vm.MessageFrame;
+import org.hyperledger.besu.ethereum.vm.OperandStack.OverflowException;
+import org.hyperledger.besu.ethereum.vm.OperandStack.UnderflowException;
 
 import org.apache.tuweni.bytes.Bytes32;
 import org.apache.tuweni.units.bigints.UInt256;
 
-public class SLoadOperation extends AbstractOperation {
+public class SLoadOperation extends AbstractFixedCostOperation {
 
   public SLoadOperation(final GasCalculator gasCalculator) {
-    super(0x54, "SLOAD", 1, 1, false, 1, gasCalculator);
+    super(0x54, "SLOAD", 1, 1, false, 1, gasCalculator, gasCalculator.getSloadOperationGasCost());
   }
 
   @Override
-  public Gas cost(final MessageFrame frame) {
-    return gasCalculator().getSloadOperationGasCost();
-  }
+  public OperationResult execute(final MessageFrame frame, final EVM evm) {
+    try {
+      if (frame.getRemainingGas().compareTo(gasCost) < 0) {
+        return oogResponse;
+      }
 
-  @Override
-  public void execute(final MessageFrame frame) {
-    final Bytes32 key = frame.popStackItem();
+      final Bytes32 key = frame.popStackItem();
+
+      final Account account = frame.getWorldState().get(frame.getRecipientAddress());
+      assert account != null : "VM account should exist";
```

Comment:
Maybe be we need to avoid using assert here ?

Reply:
I deleted it.  The account should exist already as it's an account that should have existed for the call to even be made.  If account is null we will get an NPE 2 lines later, and if it's null something has broken quite severely in the EVM implementation so native NPEs communicate the failure accurately.